### PR TITLE
Feature/652 form updates

### PR DESCRIPTION
**Pull request by nicklee**

### CODE REVIEW

**Comment by nicklee** [comment]
HI @chris-lawton,

Sorry about this one, it's pretty mental… let me know if you have any questions,

Nick

File: rca/project_styleguide/templates/patterns/pages/forms/form_page.html [hunk 1]
```diff
@@ -56,7 +56,7 @@ <h1 class="title-area__heading heading heading--display-two">
                                 {% endif %}{% if widget_type == 'textarea' %}
                                 data-focus data-focustype='textarea'
                             {% endif %}>
-                                        <div class="form-item__container">
+                                        <{% if widget_type == 'checkbox_select_multiple' %}fieldset{% else %}div{% endif %} class="form-item__container">
```
Comment (chris-lawton) [inline rca/project_styleguide/templates/patterns/pages/forms/form_page.html:59]: 👌

File: rca/project_styleguide/templates/patterns/pages/forms/form_page.html [hunk 2]
```diff
@@ -67,24 +67,31 @@ <h1 class="title-area__heading heading heading--display-two">
...
-                                                <label for="{{ field.id_for_label }}" class="form-item__label {% if field.field.required %}required{% endif %}">{{ field.label }} {% if field.field.required %} <span class="form-item__required">*</span>{% endif %}</label>
-                                                {% if widget_type == 'checkbox_select_multiple'%}<p class="form-item__instruction">Select all that apply</p>{% endif %}
+                                                <{% if widget_type == 'checkbox_select_multiple' %}legend{% else %}label for="{{ field.id_for_label }}"{% endif %} class="form-item__label {% if field.field.required %}required{% endif %}">
+                                                    {% if field_type == 're_captcha_field' %}
+                                                        reCaptcha
+                                                    {% else %}
+                                                        {{ field.label }}
+                                                    {% endif %}
+                                                    {% if field.field.required %}<span class="form-item__required">*</span>{% endif %}
```
Comment (chris-lawton) [inline rca/project_styleguide/templates/patterns/pages/forms/form_page.html:76]: should this be aria-hidden? not sure if a screen reader picks up symbols

File: rca/static_src/javascript/components/form-focus.js [hunk 3]
```diff
@@ -13,51 +13,79 @@ class FormFocus {
...
                 this.formItem.classList.remove(this.hasFocusClass);
+            } else if (this.formFieldInput === document.activeElement) {
+                this.formItem.classList.add(this.hasFocusClass);
             } else {
-                this.formItem.classList.remove(this.hasFocusClass);
-                this.formItem.classList.remove(this.hasContentClass);
+                this.formItem.classList.remove(
+                    this.hasFocusClass,
+                    this.hasContentClass,
+                );
```
Comment (chris-lawton) [inline rca/static_src/javascript/components/form-focus.js:58]: the two functions in `applyStateClass` look pretty similar. you could do something like the following for less repetition

```
applyStateClass() {
    if (this.formType === 'textarea') {
        this.doSomething(this.formFieldTextarea);
    }

    if (this.formType === 'input') {
        this.doSomething(formFieldInput);
    }
}

doSomething(el) {
    if (
        el.value &&
        el === document.activeElement
    ) {
        this.formItem.classList.add(
            this.hasContentClass,
            this.hasFocusClass,
        );
    } else if (el.value) {
        this.formItem.classList.add(this.hasContentClass);
        this.formItem.classList.remove(this.hasFocusClass);
    } else if (el === document.activeElement) {
        this.formItem.classList.add(this.hasFocusClass);
    } else {
        this.formItem.classList.remove(
            this.hasFocusClass,
            this.hasContentClass,
        );
    }
}
```

on the other hand though you may want to keep them separate so it's easier to read and more customisable in the future : )

File: rca/static_src/javascript/components/form-focus.js [hunk 4]
```diff
@@ -13,51 +13,79 @@ class FormFocus {
...
             }
         }
     }
 
+    checkFormItemStatus() {}
```
Comment (chris-lawton) [inline rca/static_src/javascript/components/form-focus.js:63]: guessing this is upcoming work?

File: rca/static_src/sass/components/_form-item.scss [hunk 5]
```diff
@@ -362,6 +373,14 @@
         @include font-size(xxs);
     }
 
+    input,
+    textarea,
+    select {
+        &:-webkit-autofill {
```
Comment (chris-lawton) [inline rca/static_src/sass/components/_form-item.scss:379]: doesn't look like there's a `:moz` equivalent for these, guessing you don't need them then?

**Comment by chris-lawton** [comment]
Few comments but can merge as is 👍

File: rca/static_src/javascript/components/form-focus.js (hunk 4 above)
Comment (nicklee) [reply rca/static_src/javascript/components/form-focus.js]: oops, it was, but then wasn't needed, thanks

File: rca/static_src/sass/components/_form-item.scss (hunk 5 above)
Comment (nicklee) [reply rca/static_src/sass/components/_form-item.scss:379]: as far as I can see mozilla no long offers autocomplete, so we don't need to worry about it

File: rca/static_src/javascript/components/form-focus.js (hunk 3 above)
Comment (nicklee) [reply rca/static_src/javascript/components/form-focus.js]: Thanks man!

File: rca/project_styleguide/templates/patterns/pages/forms/form_page.html (hunk 2 above)
Comment (nicklee) [reply rca/project_styleguide/templates/patterns/pages/forms/form_page.html]: Wasn't sure, but now you've said it screen readers will take required from the input param, so i'll add it in

**Pull request merged**